x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java

```diff
@@ -342,6 +343,15 @@ public void testAuditFilterSettingValidation() {
         e = expectThrows(IllegalArgumentException.class,
                 () -> LoggingAuditTrail.FILTER_POLICY_IGNORE_INDICES.getConcreteSettingForNamespace("filter4").get(settings4));
         assertThat(e, hasToString(containsString("invalid pattern [/no-inspiration]")));
+
+        Settings settings5 = Settings.builder()
+            .putList(prefix + "ignore_filters.filter2.users", Arrays.asList("tom", "cruise"))
+            .putList(prefix + "ignore_filters.filter2.index_privileges", Arrays.asList("read", "/foo")).build();
+        assertThat(LoggingAuditTrail.FILTER_POLICY_IGNORE_PRINCIPALS.getConcreteSettingForNamespace("filter2").get(settings5),
+            containsInAnyOrder("tom", "cruise"));
+        e = expectThrows(IllegalArgumentException.class,
+            () -> LoggingAuditTrail.FILTER_POLICY_IGNORE_INDEX_PRIVILEGES.getConcreteSettingForNamespace("filter2").get(settings5));
+        assertThat(e, hasToString(containsString("invalid pattern [/foo]")));
```

Comment:
As above, we need more testing around the validation.
We have made a decision that you cannot specify both index and cluster privileges, but we have no tests for that.